### PR TITLE
Add canned corn to corn including recipes

### PR DESCRIPTION
**Pull request by Cyrano7**

### CODE REVIEW

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -8572,7 +8578,7 @@
   "batch_time_factors":[83, 3],
   "tools": [    [ [ "hotplate", 5 ], [ "char_smoker", 5 ], [ "toolset", 5 ], [ "fire", -1 ] ],
                 [ ["rock_quern", -1], ["clay_quern", -1] ] ],
-  "components": [ [ ["corn", 2], ["irradiated_corn", 2] ] ]
+  "components": [ [ ["can_corn", 2], ["corn", 2], ["irradiated_corn", 2] ] ]
```
Comment (Rivet-the-Zombie) [inline data/json/recipes/recipe_food.json]: You can't make cornmeal from tinned sweetcorn.

Comment (Cyrano7) [inline data/json/recipes/recipe_food.json]: Ya I was on the fence about that one. If you smoked/cooked it and got the water even? Part of the recipe process is cooking it.

Comment (Rivet-the-Zombie) [inline data/json/recipes/recipe_food.json]: Nope, still needs something more like a field corn.

Comment (Cyrano7) [inline data/json/recipes/recipe_food.json]: Ya I guess your right. One sec I'll change it now.

**Comment by Cyrano7** [comment]
@Rivet-the-Zombie out now

File: data/json/recipes/recipe_food.json [hunk 2]
```diff
@@ -3913,9 +3914,9 @@
       [ "cabbage", 1 ],
       [ "irradiated_cabbage", 1 ],
       [ "sauerkraut", 2 ],
+      [ "can_corn", 1 ],
```
Comment (mugling) [inline data/json/recipes/recipe_food.json]: Moving the lines around doesn't change anything?

Comment (Cyrano7) [inline data/json/recipes/recipe_food.json]: @mugling Ya I should have written why. So they would display in recipes in the same pattern.

Comment (mugling) [inline data/json/recipes/recipe_food.json]: Ah I see, good point

**Pull request merged**